### DISCUSSION TITLE
Advice needed. Update user profile from trigger function.

### DISCUSSION THREAD

**Discussion by Maris1984** [supabase/supabase#18512] [Questions]
I have a custom public.userProfiles table where a new entry is created every time a new user signs up. 
I would like to use that public.userProfiles table to delete users from my app. I would not want to expose the updateUser method from Auth admin, since it involves exposing the service_role key, so I’m looking at a trigger on public.userProfiles table instead. It would update the user.data (set ban_duration) as soon as the user row is accordingly edited in public.userProfiles table.

However, I’m unable to find a function that I could call from a trigger that updates the user profile. Is there perhaps any other alternative and safe approach on how to achieve this?

**Comment by GaryAustin1** [comment]
Not sure what exactly the issue is.  If you want to update your public profile table you update it with .update() and have RLS set to only allow a  user to update their own role.   Then you can have an update trigger function on the profile table that when that column changes you update the auth.users column as you like.  The trigger function would need to be security definer type to do that.
